osu.Game.Rulesets.Osu/Difficulty/OsuDifficultyCalculator.cs

```diff
@@ -23,6 +24,65 @@ public OsuDifficultyCalculator(Ruleset ruleset, WorkingBeatmap beatmap)
         {
         }
 
+        protected override double strainStep() { return section_length; }
+
+        //Throws list of difficulty values at sections. First value is possesed by two sections.
+        protected override List<double> DifficultySectionRating (IBeatmap beatmap, double timeRate)
+        {
+            if (!beatmap.HitObjects.Any())
+                return new List<double>();
+
+            OsuDifficultyBeatmap difficultyBeatmap = new OsuDifficultyBeatmap(beatmap.HitObjects.Cast<OsuHitObject>().ToList(), timeRate);
+            Skill[] skills =
+            {
+                new Aim(),
+                new Speed()
+            };
+
+            double sectionLength = section_length * timeRate;
+
+            // The first object doesn't generate a strain, so we begin with an incremented section end
+            double currentSectionEnd = 2 * sectionLength;
+
+            foreach ( OsuDifficultyHitObject h in difficultyBeatmap )
```

Comment:
Whitespace style doesn't comply with project code style, remove spaces between parentheses, prefer usage of `var`